flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -1050,6 +1055,14 @@ public CreatingExecutionGraph.AssignmentResult tryToAssignSlots(
                 .orElseGet(CreatingExecutionGraph.AssignmentResult::notPossible);
     }
 
+    @Override
+    public void freeExcessiveReservedSlots() {
+        for (SlotInfo slotInfo : declarativeSlotPool.getFreeSlotsInformation()) {
+            declarativeSlotPool.freeReservedSlot(
+                    slotInfo.getAllocationId(), null, System.currentTimeMillis());
+        }
```

Comment:
Not sure what you're trying to do here. Freeing a free slot is a no-op.

Reply:
I went through the chain of differently typed callbacks; it seems that the reserved slot is already freed when the Execution is canceled (~ when the ExecutionGraph is canceled), which should make this a no-op. We're guaranteed the EG is canceled before we create a new one.

Good catch, thanks!

Reply:
(tests are still passing after removing this)